src/main/java/cubicchunks/worldgen/generator/ICubePrimer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.worldgen.generator;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+
+public interface ICubePrimer {
+
+	static final IBlockState DEFAULT_STATE = Blocks.AIR.getDefaultState();
+
+	/**
+	 * Gets a block state at the given location
+	 * 
+	 * @param x cube relative x
+	 * @param y cube relative y
+	 * @param z cube relative z
+	 * @return the block state
+	 */
+	IBlockState getBlockState(int x, int y, int z);
+
+	/**
+	 * Sets a block state at the given location
+	 * 
+	 * @param x cube relative x
+	 * @param y cube relative x
+	 * @param z cube relative x
+	 * @param state the block state
+	 */
+	void setBlockState(int x, int y, int z, IBlockState state);
```

Comment:
ICubePrimer works with local coordinates. Also BlockPos just gets in the way when it comes to generators.